src/gt4py/next/iterator/embedded.py

```diff
@@ -538,7 +538,11 @@ def execute_shift(
 ) -> MaybePosition:
     assert pos is not None
     if isinstance(tag, SparseTag):
-        current_entry = pos[tag]
+        # hack: relies on
+        # (a) Dimensions with same name are equal
+        # (b) sparse dimensions have the same name as the offset applying to them
+        sparse_dim = common.Dimension(tag)
```

Comment:
```suggestion
        # hack: relies on sparse dimensions have the same name as the offset applying to them
        sparse_dim = common.Dimension(tag, kind=common.DimensionKind.LOCAL)
```

Reply:
We already have this weird constraint, because of integration with field view lowering...